lib/Nodes/TableNode.php

```diff
@@ -98,6 +116,10 @@ public function push(?array $parts, string $line) : bool
 
     public function finalize(Parser $parser) : void
     {
+        if (count($this->headers) === count($this->data)) {
+            throw new RuntimeException('Malformed table');
```

Comment:
Thinking more about this error. When it happens, it will be hard to know where it is coming from? Thoughts @weaverryan?

Reply:
i totally agree
in the methods `finalize()` and `push()` in `\Doctrine\RST\Nodes\TableNode` we don't have access to the whole parsed table block...
wouldn't it be nice if we could have, node by node, the whole rst block, to perform some checks ?
and then, maybe print out the problematic block ?

Reply:
In `TableNode` we have access to the `Parser` and `Environment` so we should be able to keep track of what the current file is and display that in the exception...I think. Can you investigate that approach?

Reply:
yep
maybe we could even print out the malformed table itself ? 
i don't know if we could access the line number in the current file, but it would be nice to have that as well

Reply:
You should be able to print the malformed table. Should have all that data in the finalize() method. And yes, you are correct, at this point in the processing, we don't have the line number...since it is after all the lines have been parsed. I think just showing the file it is in and the contents of the malformed table will be good enough to merge this and in the future we can look at enhancements to try and detect the improperly formatted table earlier in the parsing so we can show file and line number.